sources/VT100Screen.m

```diff
@@ -848,13 +848,22 @@ - (void)clearBuffer {
 // This clears the screen, leaving the cursor's line at the top and preserves the cursor's x
 // coordinate. Scroll regions and the saved cursor position are reset.
 - (void)clearAndResetScreenPreservingCursorLine {
+    int numberOfLinesInPrompt = [iTermAdvancedSettingsModel numberOfLinesInPrompt];
```

Comment:
Rather than using an advanced setting, it would be better to figure out how tall the prompt is from shell integration. The prompt starts at the line `currentGrid_.cursor.y` when `-terminalPromptDidStart` gets called and ends at the line `currentGrid_.cursor.y` when `-terminalCommandDidStart` gets called. I would keep track of the last two values. If shell integration is not installed, then you could fall back to the advanced setting.